route/src/io/pedestal/http/route/definition/specs.clj

```diff
@@ -0,0 +1,185 @@
+; Copyright 2024 Nubank NA
+
+; The use and distribution terms for this software are covered by the
+; Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0)
+; which can be found in the file epl-v10.html at the root of this distribution.
+;
+; By using this software in any fashion, you are agreeing to be bound by
+; the terms of this license.
+;
+; You must not remove this notice, or any other, from this software.
+
+(ns io.pedestal.http.route.definition.specs
+  "Clojure spec definitions related to routing descriptions and routing tables."
+  (:require [clojure.spec.alpha :as s]
+            [clojure.string :as string]
+            [io.pedestal.interceptor :as i])
+  (:import (java.util.regex Pattern)))
+
+(defn- is-re?
+  [o]
+  (instance? Pattern o))
+
+(defn- has-meta?
+  [k]
+  (fn [v]
+    (contains? (meta v) k)))
+
+;; --- VERBOSE ROUTES ---
+
+(s/def ::verbose-routes (s/coll-of ::verbose-route))
+(s/def ::verbose-route (s/keys :opt-un [::app-name
+                                        ::scheme
+                                        ::host
+                                        ::children
+                                        ::port
+                                        ;; TODO: some of these should be sometimes required, or at least
+                                        ;; tied together.
+                                        ::path              ;; Probably required except at root
+                                        ::verbs             ;; Ditto
+                                        ::i/interceptors
+                                        ]))
+
+(s/def ::app-name keyword?)
+(s/def ::host string?)
+(s/def ::scheme #{:http :https})
+(s/def ::children (s/coll-of ::verbose-route))
+(s/def ::port (s/and integer? pos?))
+(s/def ::path string?)
```

Comment:
Since we documented it, maybe worth asserting it starts with a "/" in the spec?

Reply:
Sounds good.

Also thinking about this over the week, may move the fspec's into the specs namespace so that specs are truly optional.